tools/importers/SynchronizeTypographyTokens.ts

```diff
@@ -0,0 +1,189 @@
+						label: propertyName,
+						value: propertyValue,
+					};
+
+					if (resultJsonCss[key] === undefined) {
+						resultJsonCss[key] = [];
+					}
+					resultJsonCss[key].push(resultJsonObject);
+				}
+			}
+		}
+	}
+
+	importerVariables.resultCssAdditionalLines.forEach((item) => {
+		resultCss.push(item);
+	});
+	importerVariables.resultScssAdditionalLines.forEach((item) => {
+		resultScss.push(item);
+	});
+
+	fileWriter.arrayToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariablesCss,
+		resultCss,
+	);
+	fileWriter.arrayToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariables,
+		resultScss,
+	);
+	fileWriter.jsonToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariablesCssJson,
+		resultJsonCss,
+	);
+};
+
+const ImportTypographyTokensRaw = (binValues: ConfigFileObject, jsonTypography: any) => {
+	let resultScss: Array<ITypographyToken> = [];
+
+	for (let key in jsonTypography[importerVariables.tokensKey]) {
+		let token: ITypographyToken = fileRead.recursiveTokenReader(
+			jsonTypography[importerVariables.tokensKey][key],
+			key,
+			[],
+			importerVariables,
+		);
+		resultScss.push(token);
+	}
+
+	fileWriter.arrayToMixinFile(tokensFilesConfig.destinationPath + binValues.destination, [
+		importerVariables.importVariables,
+		...buildTypographyTokensMixins(resultScss.flat()),
+	]);
+};
+
+function buildTypographyTokensMixins(tokens: Array<ITypographyToken>): Array<string> {
+	let result: Array<string> = [];
+
+	for (let key in tokens) {
+		result.push('\n@mixin ' + tokens[key].tokenCamelCase + '() {');
+		tokens[key].attributes.forEach((attribute) => {
+			let [cssAttributePartOne, cssAttributePartTwo] = attribute.split('-');
+			result.push(
+				cssAttributePartOne +
+					'-' +
+					cssAttributePartTwo +
+					': $' +
+					importerVariables.typographyPrefix +
+					attribute +
+					';',
+			);
+		});
+		result.push('}');
+	}
+
+	return result;
+}
+
+const SynchronizeSingleBin = async (bin: ConfigFileBin) => {
+	let requestResponse = await requestForBin(bin);
+
+	ImportTypographyRaw(
+		bin.name,
+		bin.files.variablesRaw,
+		requestResponse.data.record.values.LMSDesignSystemTypography,
+		bin.isTheme,
+	);
+
+	ImportTypographyTokensRaw(
+		bin.files.tokens,
+		requestResponse.data.record.values.LMSDesignSystemTypography,
+	);
+
+	console.log('The import was successful for bin: ' + bin.id);
+};
+
+const requestForBin = async (bin: ConfigFileBin) => {
+	const requestConfig = {
+		headers: {
+			'X-Master-Key': process.env.JSON_BIN_X_MASTER_KEY,
+		},
+	};
+	const requestResponse = await axios.get(
+		tokensFilesConfig.jsonBinApiUrl + bin.id + '/latest',
+		requestConfig,
+	);
+
+	if (!requestResponse.data.record.values.LMSDesignSystemTypography) {
+		throw new TypeError('Response structure has no content!');
+	}
+
+	return requestResponse;
+};
+
+const SynchronizeTypographyTokens = async () => {
+	tokensFilesConfig.bins.forEach((bin: ConfigFileBin) => {
```

Comment:
I'm not sure if keeping the same config structure as in Colors makes sense here. Now we have iteration over bins, but we know it's very unlikely we'll need to process two JSONs

Reply:
I splitted config interface.